theano/tensor/nnet/corr_gemm.c

```diff
@@ -224,6 +217,8 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
                     bottomWidth, kH, kW, padH, padW, dH, dW,
                     (%(float_type)s*)PyArray_DATA(col)+ tid * col_stride);
             // Second, gemm
+            // Always forcing gemm to one thread here for best and stable performance.
+            %(blas_flags)s;
```

Comment:
Good point, it should restore to the previous thread number after the call to gemm as you mentioned. Looks like OpenBLAS has different way as MKL did to to so, let me verify them and update later.